File: awesomeimagepicker/src/main/java/in/myinnos/awesomeimagepicker/models/Image.java
package in.myinnos.awesomeimagepicker.models;

import android.net.Uri;
import android.os.Build;
import android.os.Parcel;

/**
 * Created by MyInnos on 03-11-2016.
 */
public class Image extends Media {

    public Image() {

    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeLong(getId());
        dest.writeString(getName());
        dest.writeString(getMimeType());
        dest.writeLong(getSize());
        dest.writeString(getUri().toString());
        /*
         * Android 11 and up can check if the media is marked as favorite
         */
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            dest.writeBoolean(isFavorite());
        }
    }

    public static final Creator<Image> CREATOR = new Creator<Image>() {
        @Override
        public Image createFromParcel(Parcel source) {
            return new Image(source);
        }

        @Override
        public Image[] newArray(int size) {
            return new Image[size];
        }
    };

    private Image(Parcel in) {
        setId(in.readLong());
        setName(in.readString());
        setMimeType(in.readString());
        setSize(in.readLong());
        setUri(Uri.parse(in.readString()));
        /*
         * Android 11 and up can check if the media is marked as favorite
         */
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            setFavorite(in.readBoolean());
        }
    }
}

File: awesomeimagepicker/src/main/java/in/myinnos/awesomeimagepicker/models/Image.kt
package `in`.myinnos.awesomeimagepicker.models

import android.net.Uri
import android.os.Build
import android.os.Parcel
import android.os.Parcelable

class Image : Media {
    constructor()

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeLong(id)
        dest.writeString(name)
        dest.writeString(uri.toString())
        dest.writeLong(size)
        dest.writeString(mimeType)
        dest.writeLong(albumId)
        dest.writeString(albumName)
        dest.writeLong(dateAddedSecond)
        /*
         * Android 11 and up can check if the media is marked as favorite
         */
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            dest.writeBoolean(isFavorite)
        }
    }

    private constructor(parcel: Parcel) {
        id = parcel.readLong()
        name = parcel.readString()
        uri = Uri.parse(parcel.readString())
        size = parcel.readLong()
        mimeType = parcel.readString()
        albumId = parcel.readLong()
        albumName = parcel.readString()
        dateAddedSecond = parcel.readLong()
        /*
         * Android 11 and up can check if the media is marked as favorite
         */
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            isFavorite = parcel.readBoolean()
        }
    }

    companion object CREATOR : Parcelable.Creator<Image> {
        override fun createFromParcel(parcel: Parcel): Image {
            return Image(parcel)
        }

        override fun newArray(size: Int): Array<Image?> {
            return arrayOfNulls(size)
        }
    }
}
